src/main/java/carleton/sysc4907/communications/ClientManager.java

```diff
@@ -25,6 +26,16 @@ public ClientManager(
         this.sendingQueue = new LinkedBlockingQueue<TargetedMessage>();
         this.sender = new TCPSender(this.sendingQueue, this.clientList, this);
 
-        new Thread(sender).start();
+        this.senderThread = new Thread(sender);
+        this.senderThread.start();
+    }
+
+    public void close() {
+        sendingQueue.clear();
+        long[] ids = clientList.getClients().keySet().stream().mapToLong(x -> (long) x).toArray();
```

Comment:
Do we need this? If I understand correctly this sends to all users, in which case we could provide an empty array of IDs